### ISSUE TITLE
Cannot share files to mobile using desktop

### ISSUE THREAD

**Issue by ReyLeonardAmorato** [andyholmes/gnome-shell-extension-gsconnect#98]
I'm trying to send a file to my mobile using my desktop but nothing happens. No notifications on the phone, no notifications on the desktop. This however, is not the case with my laptop. From my laptop I can send files successfully.

I have tried:
- disabling and enabling extension from gnome-tweak-tool
- deleting folder from nautilus and re-installing the extension
- restarted gdm and nautilus multiple times
- logged out multiple times

I also made sure required dependencies are installed.

Running debug mode on GSConnect shows:
```JS LOG: [gsconnect@andyholmes.github.io]: Share: shareUri()
JS LOG: [gsconnect@andyholmes.github.io]: LanUploadChannel.open(25b593c95e556867)
JS LOG: [gsconnect@andyholmes.github.io]: LanChannel.send(25b593c95e556867 {"id":0,"type":"kdeconnect.share.request","body":{"filename":"Screenshot from 2018-05-12 10-54-32.png"},"payloadSize":2718158,"payloadTransferInfo":{"port":1747}})
```

but it stops there, on my laptop I can see more lines showing the transfer after those.

**Comment by andyholmes** [comment]
This seems like it might be related to #26 which was fixed in v11. Are you using v11 which is still in review or v10 from the extensions website?

**Comment by ReyLeonardAmorato** [comment]
I'm using v11. I tried following the steps from the link you gave me prior to posting the issue.

**Comment by andyholmes** [comment]
I'm not sure what could be going wrong, especially if there are no errors or further debug messages. If it sends correctly from one device, it should send from another. When this line is printed:

    JS LOG: [gsconnect@andyholmes.github.io]: LanChannel.send(25b593c95e556867 {"id":0,"type":"kdeconnect.share.request","body":{"filename":"Screenshot from 2018-05-12 10-54-32.png"},"payloadSize":2718158,"payloadTransferInfo":{"port":1747}})

that means GSConnect succeeded in opening a transfer port, preparing the upload stream, sent the transfer information to the mobile device, and is now waiting for the connection to be accepted. If there were a TLS error or any other error the connection should be closed and a failed notification shown. If you have logcat on your mobile does it show any errors?

**Comment by ReyLeonardAmorato** [comment]
I don't have root on my phone unfortunately. Is there any way I can give you more info?
I should mention that my phone and desktop are on different subnets:

Laptop: 192.168.0/24
Mobile: 192.168.0/24
Desktop: 192.168.1/24

I can use every plugin from desktop > mobile and vice versa, so connections must be coming through, right? Is there any reason for the share plugin not work under these circumstances?

**Comment by andyholmes** [comment]
Well, I guess that's probably the best explanation for why it's not working. If it's possible for you to upload from your mobile to your desktop, but not your desktop to your mobile, then it's likely a routing issue.

The downloading side is given a port to connect to by the uploading side, which is usually something that's allowed (eg. passing through firewalls, etc). For the uploader though, it opens a port between 1739 and 1764 and waits for an incoming connection, which is more like running a web server; you have to allow incoming connections on that port. If your gateway is doing any port blocking or firewalling you should have a look at that and see if you can allow those ports.

**Comment by ReyLeonardAmorato** [comment]
Thanks for the ELI5. I appreciate it. I guess you can go ahead and close this as it's a routing issue from my side. On an unrelated note, I want to ask you something, is that alright? How do I contact you?

**Comment by andyholmes** [comment]
No problem, lots of issues like these never affect me so it's helpful to know how they happen to others. If it's not an issue or feature request you can send email to the address on my Github profile page.

**Comment by ReyLeonardAmorato** [comment]
I decided to try and change my network topology to see if anything will change. Now the ISP router (w/ WiFi) is connected directly to the switch alongside the file server and desktops (freeing up a NIC in my server). It has the same IP subnet as the rest of the network now. Problem described here went away and I can now send files from my desktop to mobile. Thanks for your patience man.
I'll drop in an email soon.
